controllers/argocd/util.go

```diff
@@ -1315,3 +1316,64 @@ func (r *ReconcileArgoCD) setManagedNamespaces(cr *argoproj.ArgoCD) error {
 	r.ManagedNamespaces = namespaces
 	return nil
 }
+
+func isProxyCluster() bool {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		log.Error(err, "failed to get k8s config")
+	}
+
+	// Initialize config client.
+	configClient, err := configv1client.NewForConfig(cfg)
+	if err != nil {
+		log.Error(err, "failed to initialize openshift config client")
+		return false
+	}
+
+	proxy, err := configClient.Proxies().Get(context.TODO(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		log.Error(err, "failed to get proxy configuration")
+		return false
+	}
+
+	if proxy.Spec.HTTPSProxy != "" {
+		log.Info("proxy configuration detected")
+		return true
+	}
+
+	return false
+}
+
+func getOpenShiftAPIURL() string {
```

Comment:
@iam-veeramalla @wtam2018 @jopit 
I understand this code is here as a part of the enhancement that allows using keycloak within a proxy/disconnected cluster environment
That being said, it is still another piece of openshift specific code we are inserting into the upstream operator
I was wondering if people actually use keycloak in such environments through the community operator, or if this is something we should think about moving to the gitops-operator?

Reply:
No. This cannot be moved out of the argocd operator. 

Yes, argo-cd operator can be run behind a proxy :)

Reply:
should we consider moving it to the openshift package then?

Reply:
we are talking about moving the openshift package out of this operator in that case I feel it should stay here.

Reply:
I don't think we can move the whole package out as is, the way I understand it, the existing hooks also set up some non-openshift specific role permissions (eg cluster-config permissions) so I was working on just moving the openshift specific stuff out, leaving the remaining as is